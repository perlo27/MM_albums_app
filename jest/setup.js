import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { FluidNavigator } from 'react-navigation-fluid-transitions';

Enzyme.configure({ adapter: new Adapter() });

global.fetch = require('jest-fetch-mock');

jest.mock('react-navigation-fluid-transitions', () => ({
  'Transition': 'Transition',
  'FluidNavigator': jest.fn()
}));

jest.mock('native-base', () => ({
  'Header': 'Header',
  'Body': 'Body',
  'Left': 'Left',
  'Button': 'Button',
  'Icon': 'Icon',
}));