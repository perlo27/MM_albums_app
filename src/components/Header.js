import React from 'react';
import { Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { Header as NBHeader, Left, Body, Button, Icon } from 'native-base';

import { headerHeight } from '../config';
import { navigation } from './propTypes';

export const Header = ({ title, navigation }) => (
  <NBHeader style={styles.header}>
    <Left style={styles.left}>
      <Button transparent onPress={() => navigation.goBack()}>
        <Icon name="arrow-back" />
      </Button>
    </Left>
    <Body style={styles.body}>
      <Text numberOfLines={2} style={styles.text}>
        {title}
      </Text>
    </Body>
  </NBHeader>
);

const styles = StyleSheet.create({
  header: {
    backgroundColor: 'white',
    height: headerHeight,
    paddingTop: 0,
  },
  left: {
    flex: 0,
  },
  body: {
    alignItems: 'flex-start',
    marginLeft: 5,
  },
  text: { fontWeight: 'bold' },
});

Header.propTypes = {
  title: PropTypes.string,
  navigation,
};
Header.defaultProps = {};
