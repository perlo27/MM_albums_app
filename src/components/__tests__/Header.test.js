import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Header } from '../Header';
import { navigation } from '../../../jest/moks';

const title = 'test';

describe('Header', () => {
  it('should render correctly on ios', () => {
    const tree = renderer.create(<Header navigation={navigation} title={title} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly on android', () => {
    jest.mock('Platform', () => ({
      OS: 'android',
    }));

    const tree = renderer.create(<Header navigation={navigation} title={title} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call navigate.goBack on back button press', () => {
    const wrapper = shallow(<Header navigation={navigation} title={title} />);
    wrapper.find('Button').simulate('press');
    expect(navigation.goBack).toHaveBeenCalled();
    console.log(wrapper.find('Button').simulate('press'));
  });
});
