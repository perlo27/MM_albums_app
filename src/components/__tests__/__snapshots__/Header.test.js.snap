// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly on android 1`] = `
<Header
  style={
    Array [
      Object {
        "height": 50,
        "paddingTop": 0,
      },
      false,
    ]
  }
>
  <Left
    style={
      Object {
        "flex": 0,
      }
    }
  >
    <Button
      onPress={[Function]}
      transparent={true}
    >
      <Icon
        name="arrow-back"
      />
    </Button>
  </Left>
  <Body
    style={
      Object {
        "alignItems": "flex-start",
        "marginLeft": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      test
    </Text>
  </Body>
</Header>
`;

exports[`Header should render correctly on ios 1`] = `
<Header
  style={
    Array [
      Object {
        "height": 50,
        "paddingTop": 0,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <Left
    style={
      Object {
        "flex": 0,
      }
    }
  >
    <Button
      onPress={[Function]}
      transparent={true}
    >
      <Icon
        name="arrow-back"
      />
    </Button>
  </Left>
  <Body
    style={
      Object {
        "alignItems": "flex-start",
        "marginLeft": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      test
    </Text>
  </Body>
</Header>
`;
